sound/soc/sof/loader.c

```diff
@@ -145,7 +146,7 @@ int snd_sof_parse_module_memcpy(struct snd_sof_dev *sdev,
 				block->size);
 			return -EINVAL;
 		}
-		snd_sof_dsp_block_write(sdev, sdev->mmio_bar, offset,
+		snd_sof_dsp_block_write(sdev, bar, offset,
```

Comment:
wondering if this works on Intel platforms such as Baytrail or Broadwell?

Reply:
It probably doesn't, I see the following kernel issue in CI results:

````
[    8.629125 <    0.000017>] RIP: 0010:__iowrite32_copy+0x2/0x10
[    8.629132 <    0.000007>] Code: c2 01 74 01 a4 f3 c3 a4 48 83 ea 01 eb d5 66 a5 48 83 ea 02 eb d9 66 90 e9 ab d5 56 00 90 90 90 90 90 90 90 90 90 90 90 89 d1 <f3> a5 c3 90 90 90 90 90 90 90 90 90 90 90 48 8b 47 10 48 8b 00 c3
[    8.629145 <    0.000013>] RSP: 0000:ffffb273c07e79f0 EFLAGS: 00010206
[    8.629152 <    0.000007>] RAX: 00000000000c0000 RBX: 00000000000c0000 RCX: 000000000000003c
[    8.629159 <    0.000007>] RDX: 000000000000003c RSI: ffffb273c077a028 RDI: 00000000000c0000
[    8.629165 <    0.000006>] RBP: ffffb273c077a028 R08: 00000000000000f0 R09: 00000000000000f0
[    8.629172 <    0.000007>] R10: ffffb273c07cfe38 R11: ffffffffc0582330 R12: 0000000000000000
[    8.629178 <    0.000006>] R13: 000000000000003c R14: 0000000000000000 R15: 0000000000000001
[    8.629186 <    0.000008>] FS:  00007f3b07de6680(0000) GS:ffffa1c8b4800000(0000) knlGS:0000000000000000
[    8.629194 <    0.000008>] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    8.629200 <    0.000006>] CR2: 00000000000c0000 CR3: 000000007807c000 CR4: 00000000001006f0
[    8.629206 <    0.000006>] Call Trace:
[    8.629222 <    0.000016>]  sof_block_write+0x34/0xcd0 [snd_sof]
[    8.629234 <    0.000012>]  snd_sof_parse_module_memcpy+0x103/0x1c0 [snd_sof]
[    8.629246 <    0.000012>]  ? snd_sof_load_firmware+0x50/0x50 [snd_sof]
[    8.629255 <    0.000009>]  snd_sof_load_firmware_memcpy+0x128/0x230 [snd_sof]
```

Reply:
@plbossart thanks for pointing this. I now remember about this patch: https://github.com/thesofproject/linux/pull/604/commits/1560d7aec52bf556e7d57ee5ae18165f3bf7ee39 I think that was trying to fix this kind of issues. Will have a more detailed look.